File: SparkXAdSDKDemo/SparkXAdDemo/AdListViewController.m
//
//  AdListViewController.m
//  SparkXAdDemo
//
//  Created by Wei yang on 2021/2/2.
//  Copyright © 2021 SparkX. All rights reserved.
//

#import "AdListViewController.h"

#define Title @"title"
#define ClassTitle @"class"

@interface AdListViewController ()<UITableViewDelegate,UITableViewDataSource>
@property (nonatomic, strong) NSMutableArray *listArray;

@end

@implementation AdListViewController

- (NSMutableArray *)listArray
{
    if (!_listArray) {
        _listArray = [NSMutableArray array];
        [_listArray addObject:@{Title:@"NativeAd",ClassTitle:@"NativeAdViewController"}];
        [_listArray addObject:@{Title:@"SplashAd",ClassTitle:@"SplashAdViewController"}];
        [_listArray addObject:@{Title:@"BannerAd",ClassTitle:@"BannerAdViewController"}];

        [_listArray addObject:@{Title:@"InterstitialAd",ClassTitle:@"InterstitialAdViewController"}];
    }
    return _listArray;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.title = @"SparkX demo";
    
    UITableView *listView = [[UITableView alloc] initWithFrame:self.view.bounds];
    listView.delegate =  self;
    listView.dataSource = self;
    listView.tableFooterView = [[UIView alloc] init];
    listView.backgroundColor = [UIColor whiteColor];
    [self.view addSubview:listView];
    
    UIButton *leftbutton=[[UIButton alloc]initWithFrame:CGRectMake(0, 0, 100, 20)];

    //[leftbutton setBackgroundColor:[UIColor blackColor]];
    [leftbutton setTitle:@"v1.0.0" forState:UIControlStateNormal];
    [leftbutton setTitleColor:UIColor.blackColor forState:UIControlStateNormal];
//    [leftbutton addTarget:self action:@selector(devModel:) forControlEvents:UIControlEventTouchUpInside];
    UIBarButtonItem *leftitem=[[UIBarButtonItem alloc]initWithCustomView:leftbutton];

    self.navigationItem.leftBarButtonItem=leftitem;
}



#pragma mark UITableViewDelegate,UITableViewDataSource
- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.listArray.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"abc"];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"abc"];
    }
    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    cell.textLabel.text = [self.listArray[indexPath.row] objectForKey:Title];
    
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    NSString *classStr = [self.listArray[indexPath.row] objectForKey:ClassTitle];
    Class vcClass = NSClassFromString(classStr);
    UIViewController *vc = [[vcClass alloc] init];
    vc.title = [self.listArray[indexPath.row] objectForKey:Title];
    [self.navigationController pushViewController:vc animated:YES];
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: SparkXAdDemo-Swift/SparkXAdDemo_Swift/AdListViewController.swift
//
//  ViewController.swift
//  SparkXAdDemo-Swift
//
//  Created by Wei yang on 2021/3/16.
//

import UIKit

class AdListViewController: UIViewController,UITableViewDelegate,UITableViewDataSource {
    
    
    private let _items =  [
        ["title":"Native","class":"NativeAdViewController"],
        ["title":"Banner","class":"BannerAdViewController"],
        ["title":"Splash","class":"SplashAdViewController"],
        ["title":"Interstitial","class":"InterstitialAdViewController"],

    ];
    
    
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return _items.count;
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        var cell = tableView.dequeueReusableCell(withIdentifier: "abcCell")
        if (cell==nil) {
            cell = UITableViewCell();
            cell?.accessoryType = .disclosureIndicator;
            cell?.selectionStyle = .none;
        }
        let subData = _items[indexPath.row];
        cell?.textLabel?.text = subData["title"];
        return cell!;
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true);
        let data = _items[indexPath.row];
        let name:String = data["class"]!
        if (name.count > 0){
            let namespace = Bundle.main.infoDictionary!["CFBundleExecutable"] as! String
            let cls:AnyObject = NSClassFromString(namespace + "." + name)!
            let viewControllerClass = cls as! UIViewController.Type
            let viewController = viewControllerClass.init()
            self.navigationController?.pushViewController(viewController, animated: true)

        }
        
    }
    


    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        let tableView = UITableView(frame: self.view.bounds, style: .grouped)
            tableView.delegate = self
            tableView.dataSource = self
        
        self.view.addSubview(tableView);
    }


}

